libs/ui/src/components/activity-section-list/activity-section-list.tsx

```diff
@@ -45,16 +48,53 @@ const renderSectionHeader = (item: { section: SectionListData<ActivityData, Sect
 
 const emptyIconSize = getCustomSize(isWeb ? 30 : 36);
 
-export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, tokenAddress = '' }) => {
+export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, filterType, tokenAddress = '' }) => {
   const [offsetY, setOffsetY] = useState(0);
-  const { activity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+  const { activity: allActivity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+
+  const [activity, setActivity] = useState(allActivity);
 
   const handleFetchData = () => {
     if (offsetY === 0) {
       fetch(0);
     }
   };
 
+  const filterActivity = (condition: (activity: ActivityData) => boolean) => {
+    setActivity(
+      allActivity
+        .map(item => ({
+          title: item.title,
+          data: item.data.filter(activity => condition(activity))
+        }))
+        .filter(item => item.data.length)
+    );
+  };
+
+  useEffect(() => {
+    if (isDefined(filterType)) {
+      switch (filterType) {
+        case ActivityFilterEnum.Send: {
+          return filterActivity(activity => activity.type === TransactionTypeEnum.Send);
+        }
+        case ActivityFilterEnum.Receive: {
+          return filterActivity(activity => activity.type === TransactionTypeEnum.Receive);
+        }
+        case ActivityFilterEnum.ContractInteraction: {
+          return filterActivity(activity => activity.type === TransactionTypeEnum.ContractInteraction);
+        }
+        case ActivityFilterEnum.Collectibles: {
+          return filterActivity(activity => activity.isCollectible ?? false);
+        }
+        case ActivityFilterEnum.AllActivity: {
+          return setActivity(allActivity);
+        }
+      }
+    } else {
+      setActivity(allActivity);
+    }
+  }, [filterType, allActivity]);
```

Comment:
create a util that will extract filtering logic from the component.  
Additionally, it's look like you could use `useMemo` instead of `useState` for `activity` (this's will remove 1 redundant rerender)

```
const filterActivity = (activity, filterType) => activity.filter....


...

const activity = useMemo(() => filterActivity(activity, filterType), [activity, filterType]);

```